content/docs/fundamentals/converting/resolving-errors@pt_BR.md

```diff
@@ -0,0 +1,350 @@
+</pre>
+
+Todo documento AMP precisa ter um link que faça referência à respectiva versão "canônica".  Veja mais sobre o que é uma página canônica e quais são as diferentes abordagens para vincular o conteúdo canônico na etapa deste tutorial sobre [como fazer com que a página seja detectável](/pt_br/docs/fundamentals/converting/discoverable.html).
+
+Neste tutorial, o artigo HTML original que estamos convertendo será a página canônica.
+
+**Adicione** o código a seguir à tag `<meta charset="utf-8" />`:
+
+```html
+<link rel="canonical" href="/article.html">
+```
+
+{% call callout('Observação', type='note') %}
+É possível criar uma página AMP canônica independente. Nesse caso, o link canônico ainda é necessário, mas precisa levar ao próprio artigo AMP:
+
+```html
+<link rel="canonical" href="article.amp.html">
+```
+{% endcall %}
+
+Agora, **atualize** a página. Embora ainda haja muitos erros para corrigir, o erro do link canônico não está mais presente.
+
+## Especificar o atributo AMP
+
+A AMP exige um atributo no elemento raiz `<html>` das páginas para declarar que elas são documentos AMP.
+
+<pre class="error-text">
+O atributo "⚡" obrigatório está ausente na tag "html ⚡ for top-level html".
+A tag obrigatória "html ⚡ for top-level html" está ausente ou incorreta.
+</pre>
+
+Para corrigir os erros acima, basta adicionar o atributo `⚡` à tag `<html>` da seguinte forma:
+
+```html
+<html ⚡ lang="en">
+```
+
+Agora, atualize a página e verifique se os erros desapareceram.
+
+{% call callout('Observação', type='note') %}
+A abordagem recomendada é especificar `⚡`, mas também é possível usar o atributo `amp` no lugar do atributo `⚡`. Veja este exemplo:
+
+```html
+<html amp lang="en">
+```
+{% endcall %}
+
+## Especificar uma janela de visualização
+
+Agora, vejamos o seguinte erro:
+
+<pre class="error-text">
+A tag obrigatória "meta name=viewport" está ausente ou incorreta.
+</pre>
+
+A AMP exige a definição de `width` e `minimum-scale` para a janela de visualização. Esses valores precisam ser definidos como `device-width` e `1`, respectivamente. A janela de visualização é uma tag comum incluída no elemento `<head>` de uma página HTML.
+
+Para corrigir o erro na janela de visualização, adicione o seguinte snippet HTML à tag `<head>`:
+
+```html
+<meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
+```
+
+Os valores especificados para `width` e `minimum-scale` são obrigatórios na AMP. A definição de `initial-scale` não é obrigatória, mas é recomendada e costuma ser incluída no desenvolvimento da Web para dispositivos móveis. Leia mais sobre a janela de visualização e o design responsivo em [Defina a janela de visualização](https://developers.google.com/speed/docs/insights/ConfigureViewport).
+
+Assim como antes, **atualize** a página e verifique se o erro desapareceu.
+
+## Substituir folhas de estilo externas
+
+O erro a seguir está relacionado com o uso de folhas de estilo:
+
+<pre class="error-text">
```

Comment:
All of these code blocks need to be in EN, so please update with EN translation (https://www.ampproject.org/docs/fundamentals/converting/resolving-errors).